### PR TITLE
 Add summary report to diagnostic bundle

### PR DESCRIPTION
**Pull request by janisz**
This PR add new state `Failed` to diagnostics bundles. And use this state in `report.json` to indicate if bundle was successfully generated on specific node. Report is included in diagnostic bundle.

### CODE REVIEW

**Comment by janisz** [comment]
@drewvanstone I just used this mocking approach with a difference I do not check if function was invoked

**Comment by janisz** [comment]
:wave: @makkes  Can you review, please?

File: api/rest/new_mock_client_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package rest
+
+import "context"
+
+type AlternativeMockClient struct {
```
Comment (makkes) [inline api/rest/new_mock_client_test.go]: Why the `Alternative` here?

Comment (janisz) [reply api/rest/new_mock_client_test.go]: We already have `MockClient` I don't know how to name this one. Maybe I can alter the code to use teststify

File: api/rest/coordinator.go [hunk 2]
```diff
@@ -58,6 +60,16 @@ func NewParallelCoordinator(client Client, interval time.Duration, workDir strin
 	}
 }
 
+type bundleReport struct {
+	ID    string                      `json:"id"`
+	Nodes map[string]bundleNodeReport `json:"nodes"`
+}
+
+type bundleNodeReport struct {
```
[the remark this answers is not in the document]
Comment (janisz) [reply api/rest/coordinator.go]: Fixed

File: api/rest/new_mock_client_test.go (hunk 1 above)
Comment (janisz) [reply api/rest/new_mock_client_test.go]: Renamed

**Pull request merged**